models/user/email_address.go

```diff
@@ -41,6 +41,7 @@ func (err ErrEmailCharIsNotSupported) Error() string {
 }
 
 // ErrEmailInvalid represents an error where the email address does not comply with RFC 5322
+// or has a leading '-' character
```

Comment:
I'm not sure - should be the only forbidden leading character, because the issue is people using sendmail to send their email, and the interface with it is prone to turning arguments into RCEs

Reply:
update, or just apply the strict filtering if the sendmail interface is used?

Reply:
Assuming that sendmail implementations are a black box, potentially any character in an e-mail address could be dangerous. Realistically, though, only the dash character will generate problems. I am not aware of any POSIX-style command line parser that treats any character beyond '-' as special (well, except for 2nd-level 'key:value' mappings such as seen in ffmpeg or mplayer... but that doesn't really apply here).

Performing the check only when sendmail is used might work, but will likely complicate the code. I would suggest that in that case @silverwind is correct, and sendmail should be scratched in favor of native SMTP (getting rid of the check, altogether).

Reply:
Yes, we should definitely evaluate dropping sendmail support if the arguments can not be safely escaped.

Reply:
I suspect we should just have a few options:

1) Allow everything but initial `-`
2) Only allow safe characters

This can be an option on the sendmail configuration and then the problem is no longer ours to solve and it becomes the system admins job to deal with it.

Reply:
I'm not quite sure what you mean with `2.`.
I'd change `2.` a bit (`1.` is fine):
`2. Allow every valid email, leave it up to the admin to allow sendmail configurations`
`3. Allow every valid email, drop sendmail support`